esp-radio/src/wifi/mod.rs

```diff
@@ -3371,6 +3371,10 @@ impl WifiController<'_> {
     ///
     /// This function will wait for the Wi-Fi controller to stop before returning.
     pub async fn stop_async(&mut self) -> Result<(), WifiError> {
+        if !self.is_started()? {
```

Comment:
I _think_ I was considering a similar approach using `is_started` but didn't submit a PR based on that - I just don't remember my concerns.

Maybe because there is a potential race - i.e. there is a start operation in progress but it didn't made it to the state where the driver emits the `Started` event.

On the other hand we probably have a few more situations like that - (as far as I know) the driver is using a queue for the commands but we look at the events of the executed commands.

This will mitigate the problem in the original issue but we probably should rethink the way we deal with events (touches #4364 but I _think_ it also affects how we deal with events in async)

Not sure what to do here, then. Maybe add a comment and merge it?